### DISCUSSION TITLE
difficulty generating network traffic and geolocation

### DISCUSSION THREAD

**Discussion by gerichkoeke** [akvorado/akvorado#1077] [Q&A]
Good afternoon, I'm using Akvorado to collect flow from Huawei NE8000 and Cisco ASR-1000. For Cisco, I used the configuration as shown in the documentation, but I cannot in any way make the network graphs and Top source As and Top and top source countries work.

follow the configuration of my inlet.yaml file

---
kafka:
   compression-codec: zstd
geoip:
   optional: true
   # When running on Docker, these paths are inside the container. By default,
   # IPinfo databases are used. (https://ipinfo.io/)
   asn-database: /usr/share/GeoIP/asn.mmdb
   geo-database: /usr/share/GeoIP/country.mmdb
   # If you want to use MaxmindDB, check `.env`, `docker-compose-maxmind.yml` and
   # update these paths:
   #asn-database: /usr/share/GeoIP/GeoLite2-ASN.mmdb
   #geo-database: /usr/share/GeoIP/GeoLite2-Country.mmdb
metadata:
   workers: 2
   providers:
     type: snmp
     communities:
       ::/0: AoWinRoC#12345
flow:
   inputs:
     - type: udp
       decoder: netflow
       listen: :2055
       workers: 6
       receive-buffer: 10485760
     - type: udp
       decoder: sflow
       listen: :6343
       workers: 6
       receive-buffer: 10485760
core:
   override-sampling-rate: 1000
   workers: 2
   exporter-classifiers:
     - ClassifySiteRegex(Exporter.Name, "^[^-]+-([^-]+)-", "$1")
     - ClassifyRegionRegex(Exporter.Name, "^([^-])+-[^-]+-", "$1")
     - ClassifyTenant("AS14457")
     - ClassifyRole("br")
   interface-classifiers:
     - |
       ClassifyConnectivityRegex(Interface.Description, "^(?i)(transit|pni|ppni|ix):? ", "$1") &&
       ClassifyProviderRegex(Interface.Description, "^\\S+?\\s(\\S+)", "$1") &&
       ClassifyExternal()
     - ClassifyConnectivityRegex(Interface.Description, "^(?i)(pppoe|cust|core):? ", "$1") &&
       ClassifyInternal()

Could you help me, I've been trying to make it work for about a week now and I've run out of ideas.
I'm using the latest version of the Docker configuration, and I've also already configured daemon.json for Docker configuration.

**Comment by gerichkoeke** [comment]
I have already used the 2 configurations, both GeoIP, updated the token and also created a token for maxmind, none of which can obtain the information in a graph.

**Comment by gerichkoeke** [comment]
I also receive these messages in the log

duration":0.353828,"time":"2024-02-03T16:30:47Z","caller":"akvorado/common/httpserver/root.go:88","module":"akvorado/common/httpserver","message":"HTTP request"}
akvorado-inlet-1            | {"level":"error","error":"IPFIX IPFIX [version:10 type:Decode obsDomainId:256: templateId:278] Error template not found","exporter":"172.29.128.254","time":"2024-02-03T16:30:47Z","caller":"akvorado/inlet/flow/decoder/netflow/root.go:198","module":"akvorado/inlet/flow/decoder/netflow","message":"error while decoding NetFlow/IPFIX"}
akvorado-inlet-1            | {"level":"error","error":"IPFIX IPFIX [version:10 type:Decode obsDomainId:256: templateId:278] Error template not found","exporter":"172.29.128.254","time":"2024-02-03T16:30:47Z","caller":"akvorado/inlet/flow/decoder/netflow/root.go:198","module":"akvorado/inlet/flow/decoder/netflow","message":"error while decoding NetFlow/IPFIX"}
akvorado-inlet-1            | {"level":"error","error":"IPFIX IPFIX [version:10 type:Decode obsDomainId:512: templateId:279] Error template not found","exporter":"172.29.128.254","time":"2024-02-03T16:30:48Z","caller":"akvorado/inlet/flow/decoder/netflow/root.go:198","module":"akvorado/inlet/flow/decoder/netflow","message":"error while decoding NetFlow/IPFIX"}

**Comment by gerichkoeke** [comment]
Hi Vicent, I finally had success with collecting the mote, but I'm having difficulty collecting the Huawei interfaces, in the case of mote we have vlan but it only uses one port. On Huawei we have an Eth-Trunk, could you tell me how I can collect it in this case?

When I try to use discussion configuration 384, I receive the following error message:

  error decoding 'Inlet[0].Core.InterfaceClassifiers[0]': cannot compile interface classifier rule "Interface.Description matches \"MegaIX NNI\" && (\nInterface.VLAN == 10 && SetDescription(\"IX: Amazon\ ") ||\nInterface.VLAN == 20 && SetDescription(\"IX: Twitch\") ||\nInterface.VLAN == 901 && SetDescription(\"Transit: Vocus\") ||\nSetDescription(\"Customer : vlan \" + Interface.VLAN))\n": invalid operation: + (mismatched types string and uint16) (5:34)
akvorado-orchestrator-1 | | SetDescription("Customer: vlan " + Interface.VLAN))

**Reply by vincentbernat** [reply]
Use `"Customer: vlan " + string(Interface.VLAN)`

**Comment by gerichkoeke** [comment]
Thanks for the feedback, now it is collecting information from the interfaces.

**Comment by gerichkoeke** [comment]
Another question, with the maxmind API is it possible to enable asn, country and city geolocation? Or it is only possible to use country and asn, or asn and city.

**Comment by gerichkoeke** [comment]
Good evening, I did a new test using the model you sent.

 error decoding 'Inlet[0].Core.InterfaceClassifiers[0]': cannot compile interface classifier rule "Interface.Description matches \"MegaIX NNI\" && (\nInterface.VLAN == 10 && SetDescription(\"IX: Amazon\") ||\nInterface.VLAN == 20 && SetDescription(\"IX: Twitch\") ||\nInterface.VLAN == 901 && SetDescription(\"Transit: Vocus\") ||\nSetDescription(\"Customer: vlan \" + String(Interface.VLAN))) &&\nClassifyProviderRegex(Interface.Description, \"^\\\\S+?\\\\s(?:to )?(\\\\S+)\", \"$1\") &&\nClassifyExternal()\n": unknown name String (5:36)
akvorado-orchestrator-1     |  | SetDescription("Customer: vlan " + String(Interface.VLAN))) &&


  interface-classifiers:
    - |
      Interface.Description matches "MegaIX NNI" && (
      Interface.VLAN == 10 && SetDescription("IX: Amazon") ||
      Interface.VLAN == 20 && SetDescription("IX: Twitch") ||
      Interface.VLAN == 901 && SetDescription("Transit: Vocus") ||
      SetDescription("Customer: vlan " + String(Interface.VLAN))) &&
      ClassifyProviderRegex(Interface.Description, "^\\S+?\\s(?:to )?(\\S+)", "$1") &&
      ClassifyExternal()
    - ClassifyConnectivityRegex(Interface.Description, "^(?i)(cust|core):? ", "$1") &&
      ClassifyInternal()
    - ClassifyInternal()

**Reply by vincentbernat** [reply]
It's `string`, not `String`.
